zulipterminal/ui_tools/boxes.py

```diff
@@ -587,6 +636,11 @@ def keypress(self, size: urwid_Size, key: str) -> Optional[str]:
         elif is_command_key("SAVE_AS_DRAFT", key):
             if not self.msg_edit_id:
                 if self.to_write_box:
+                    all_valid = self.tidy_valid_recipients_and_notify_invalid_ones(
+                        self.to_write_box
+                    )
+                    if not all_valid:
+                        return key
```

Comment:
These 2 extra checks (draft/send)  are still part of this commit; you don't mention this change in the commit text either, but it'd probably come across as an "also applies checking to x and y", which indicates it could go into another commit.